daemon/src/rollover_taker.rs

```diff
@@ -29,73 +27,88 @@ use futures::future;
 use std::time::Duration;
-use time::OffsetDateTime;
 use xtra::prelude::MessageChannel;
 use xtra_productivity::xtra_productivity;
 
 pub const MAX_ROLLOVER_DURATION: Duration = Duration::from_secs(4 * 60);
 
 pub struct Actor {
-    cfd: Cfd,
+    id: OrderId,
     n_payouts: usize,
     oracle_pk: schnorrsig::PublicKey,
-    timestamp: Timestamp,
     maker: xtra::Address<connection::Actor>,
     get_announcement: Box<dyn MessageChannel<GetAnnouncement>>,
-    projection: xtra::Address<projection::Actor>,
-    on_completed: Box<dyn MessageChannel<RolloverCompleted>>,
+    process_manager: xtra::Address<process_manager::Actor>,
     on_stopping: Vec<Box<dyn MessageChannel<Stopping<Self>>>>,
     rollover_msg_sender: Option<UnboundedSender<RollOverMsg>>,
     tasks: Tasks,
+
+    // TODO: It's a weird indirection to have the params and dlc in here
+    //  We do this, because it is "ephemeral" and we cannot easily retrieve it from the Cfd in the
+    // context of a rollover.  We could potentially have a "current rollover" in the Cfd itself
+    // that is set upon RolloverStarted. Not sure that's better.
+    rollover_params: Option<RolloverParams>,
+    dlc: Option<Dlc>,
+    db: sqlx::SqlitePool,
 }
 
 impl Actor {
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
-        (cfd, n_payouts): (Cfd, usize),
+        id: OrderId,
+        n_payouts: usize,
         oracle_pk: schnorrsig::PublicKey,
         maker: xtra::Address<connection::Actor>,
         get_announcement: &(impl MessageChannel<GetAnnouncement> + 'static),
-        projection: xtra::Address<projection::Actor>,
-        on_completed: &(impl MessageChannel<RolloverCompleted> + 'static),
+        process_manager: xtra::Address<process_manager::Actor>,
         (on_stopping0, on_stopping1): (
             &(impl MessageChannel<Stopping<Self>> + 'static),
             &(impl MessageChannel<Stopping<Self>> + 'static),
         ),
+        db: sqlx::SqlitePool,
     ) -> Self {
         Self {
-            cfd,
+            id,
             n_payouts,
             oracle_pk,
-            timestamp: Timestamp::now(),
             maker,
             get_announcement: get_announcement.clone_channel(),
-            projection,
-            on_completed: on_completed.clone_channel(),
+            process_manager,
             on_stopping: vec![on_stopping0.clone_channel(), on_stopping1.clone_channel()],
             rollover_msg_sender: None,
             tasks: Tasks::default(),
+            rollover_params: None,
+            dlc: None,
+            db,
         }
     }
 
-    async fn propose(&self, this: xtra::Address<Self>) -> Result<()> {
-        tracing::trace!(order_id=%self.cfd.id(), "Proposing rollover");
+    async fn propose(&mut self, this: xtra::Address<Self>) -> Result<()> {
+        let mut conn = self.db.acquire().await?;
+        let cfd = load_cfd(self.id, &mut conn).await?;
+
+        let (event, (rollover_params, dlc)) = cfd.start_rollover_taker()?;
```

Comment:
would it be possible to only retrieve the `(rollover_params, dlc)` after acceptance? do we need it earlier?

Reply:
I'll take a look tomorrow!

Reply:
Good catch, I think it makes a lot more sense.